pilot/pkg/bootstrap/istio_ca.go

```diff
@@ -123,27 +123,48 @@ type CAOptions struct {
 	Namespace   string
 }
 
+// EnableCA returns whether CA functionality is enabled in istiod.
+// The logic of this function is from the logic of whether running CA
+// in RunCA(). The reason for moving this logic from RunCA into EnableCA() is
+// to have a central consistent endpoint to get whether CA functionality is
+// enabled in istiod. EnableCA() is called in multiple places.
+func (s *Server) EnableCA() bool {
+	if s.kubeClient == nil {
+		// No k8s - no self-signed certs.
+		// TODO: implement it using a local directory, for non-k8s env.
+		log.Warna("kubeclient is nil; disable the CA functionality", JWTPath)
+		return false
+	}
+	iss := trustedIssuer.Get()
+	if _, err := ioutil.ReadFile(JWTPath); err != nil {
+		// for debug we may want to override this by setting trustedIssuer explicitly
+		if iss == "" {
```

Comment:
@costinm The logic in EnableCA() is from RunCA(). I am uncertain why CA functionality is disabled if JWT read fails and the issuer is empty. Can you explain the logic of this code?